### ISSUE TITLE
breqn latex package support - KaTeX

### ISSUE THREAD

**Issue by balakrishnan1978** [KaTeX/KaTeX#2005]
KaTeX is support `breqn` LaTeX Package? I would like to auto break equation without any manual work (like \\, or \cr). Otherwise how to `word-wrap` the equation? Please advise.

```\begin{dmath}  Q(\lambda,\hat{\lambda}) = -\frac{1}{2} P{(O \mid \lambda )} \sum_s \sum_m \sum_t \gamma_m^{(s)} (t) \left( n \log(2 \pi ) + \log \left| C_m^{(s)} \right| + \left( \mathbf{o}_t - \hat{\mu}_m^{(s)} \right) ^T C_m^{(s)-1} \left(\mathbf{o}_t - \hat{\mu}_m^{(s)}\right) \right)\end{dmath}```
KaTeX Version   : 0.10.2
Device          : Desktop
OS              : Ubuntu Version 16.04
Browser         : Chrome
Version         : 75.0.3770.80

**Comment by edemaine** [comment]
KaTeX supports automatic line breaking in inline math (`$...$`) but not display math (`$$...$$`), as in LaTeX.  Would `$\displaystyle ...$` work for your purposes?  That's currently the best workaround, as suggested in #1824.

There was discussion around supporting the `autobreak` package in #1023. [`breqn`](http://mirrors.concertpass.com/tex-archive/macros/latex/contrib/breqn/breqn.pdf) seems like another option to consider, though it looks substantially more complicated (not exactly automatic to use).

**Comment by hanzhihua72** [comment]
Would it be possible to add an option for KaTeX to interpret all display math modes $$...$$ as $\displaystyle ... $? It would be super convenient, without having to go back and change everything. You could just paste everything from LaTeX right into KaTeX, and doesn't seem too hard to code.

**Comment by balakrishnan1978** [comment]
Dear edemaine: I had tried your solution and not get proper output. Also how to use $\displaystyle ...$ in display math. Could you please send me one sample HTML file please? How to use \begin{autobreak} in KaTeX? I had tried the solution last few days and not get proper output. Could you please help me?

**Comment by balakrishnan1978** [comment]
@Edemaine: I have waiting for your reply last few days. Could you please advise how to achieve this. Waiting for your favourable reply.

**Comment by edemaine** [comment]
@balakrishnan1978 [Here's a demo of automatic line breaking with `$\displaystyle ...$`](https://codepen.io/edemaine/pen/WNeqPMg).

`autobreak` isn't supported by KaTeX; it was merely being discussed as something we could support.  Sorry for the confusion.

**Comment by balakrishnan1978** [comment]
@edemaine: Thanks for your reply. It's possbile after math operator line break is possible in KaTeX? Example like (+, -(minus), x(multiplication), /, % etc...)? No Need autobreak. But i will fix the text width percentage in the CSS file. It is possible? Waiting for your favourable reply.

**Comment by edemaine** [comment]
@balakrishnan1978 Yes, see https://katex.org/docs/supported.html#line-breaks which explains the line breaking behavior (same as LaTeX).

It's difficult to tell what you need in terms of CSS, but that might be better asked (with details about what behavior you want) on StackOverflow.

**Comment by balakrishnan1978** [comment]
@edemaine: when using \left and \right in Inline Equation line breaking is not working. Could you please advise how to solve this problem.

Example:
$\displaystyle \left(a+b+c+d+e+f+g+h+i+j+k+l+m+n+o+p+q+r+s+t+u+v+w+x+y+z+\frac{a}{b}+\frac{c}{d}+\frac{e}{f}+\frac{g}{h}+\frac{i}{j}+\frac{k}{l}+\frac{m}{n}+\frac{o}{p}+\frac{q}{r}+\frac{s}{t}+\frac{u}{v}+\frac{w}{x}+\frac{y}{z}\right)$

StackOverflow question tag is: https://stackoverflow.com/questions/58285622/inline-equation-line-breaking-with-using-left-and-right-command

**Comment by edemaine** [comment]
Like LaTeX, line breaking only happens at the top level of an equation, whereas `\left`/`\right` form a nested group. This is expected behavior, and unlikely to change.

**Comment by balakrishnan1978** [comment]
@edemaine: I have not received any answers from StackOverflow and i have doing R&D work and let you know updations shortly. Currently linebreak is working after `math operator`. But i have needed before linebreak with one `&emsp;` before `mop`. How to achieve this? Please find the screenshot for your reference. 
![line-break-after-mathopr](https://user-images.githubusercontent.com/5443482/66485025-b35a3080-eac5-11e9-9eb2-a16a1031193f.png)

**Comment by edemaine** [comment]
@balakrishnan1978 You can add the ability for a line break to happen anywhere *at the top level* using `\allowbreak`.  So you want to allow it before an operator, you can write e.g. `x \allowbreak + y`.  (But as before this will only work at the top level, so not inside a `\left ... \right` group.)
